### PR TITLE
OSD-16604: Add managed script wait flag until the job is finished

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
### What type of PR is this?

feature

### What this PR does / Why we need it?

Wait for job to be finished. if the wait flag specified
```
ocm backplane managedjob create SREP/example -w -p "var1=test"

Job openshift-job-aaa created.

Waiting for openshift-job-aaa  to be finished .....
Done

```

### Which Jira/Github issue(s) does this PR fix?

_Resolves #_
https://issues.redhat.com/browse/OSD-16604

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 1]
```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
...
 }
 
-func runCreateManagedJob(cmd *cobra.Command, args []string) error {
-	// ======== Parsing Flags ========
-	// Params flag
-	arr, err := cmd.Flags().GetStringArray("params")
+// runCreateManagedJob creates managed job in the specific cluster
+func runCreateManagedJob(cmd *cobra.Command, args []string) (err error) {
+
+	// init params and validate
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:59]:
```suggestion
	// initialize and validate parameters
```

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 2]
```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
...
+	err = initParams(cmd, args)
+
 	if err != nil {
 		return err
 	}
 
-	parsedParams, err := utils.ParseParamsFlag(arr)
+	// ======== Initialise backplaneURL ========
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:66]:
```suggestion
	// ======== Initialize backplaneURL ========
```

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 3]
```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
...
-	clusterID := bpCluster.ClusterID
+	return nil
+}
 
-	if urlFlag != "" {
-		backplaneHost = urlFlag
-	}
-	// ======== Parsing Args ========
-	if len(args) < 1 {
+// initParams initialise params and validate it
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:113]:
```suggestion
// initParams initialize parameters and validate them
```

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 4]
```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
...
+func initParams(cmd *cobra.Command, argv []string) (err error) {
+	// validate job canonical name
+	if len(argv) < 1 {
 		return fmt.Errorf("please input canonical-name of script as an argument. Please refer to \"ocm-backplane script list\"")
 	}
-	// It is always 1 in length, enforced by cobra
-	canonicalNameArg := args[0]
 
+	// init job cononical name
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:120]:
```suggestion
	// init job canonical name
```

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 5]
```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
...
 	if err != nil {
 		return err
 	}
 
-	cj := BackplaneApi.CreateJobJSONRequestBody{
-		CanonicalName: &canonicalNameArg,
+	return nil
+}
+
+// createJob initiates the job creation in a specific cluster and returns the job info
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:145]:
```suggestion
// createJob initializes the job creation in a specific cluster and returns the job info
```

File: cmd/ocm-backplane/managedJob/createManagedJob.go [hunk 6]
```diff
@@ -22,112 +36,182 @@ func newCreateManagedJobCmd() *cobra.Command {
...
+			return false, err
+		}
+
+		formatJobResp, err := BackplaneApi.ParseGetRunResponse(jobResp)
+
+		if err != nil {
+			return false, err
+		}
+
+		if *formatJobResp.JSON200.JobStatus.Status == "Succeeded" {
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob.go:208]: We should exit and print if the job is failed as well?

File: cmd/ocm-backplane/managedJob/createManagedJob.go (hunk 2 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/managedJob/createManagedJob.go:66]: right ..time to change my British EN lang pack to USA EN lang pack :)

File: cmd/ocm-backplane/managedJob/createManagedJob.go (hunk 6 above)
Comment (samanthajayasinghe) [reply cmd/ocm-backplane/managedJob/createManagedJob.go]: fixed. 
https://github.com/openshift/backplane-cli/blob/d30d217556c51de11f190bdb0bd4662d29e52a0e/cmd/ocm-backplane/managedJob/createManagedJob.go#L213

File: cmd/ocm-backplane/managedJob/createManagedJob_test.go [hunk 7]
```diff
@@ -126,5 +156,73 @@ var _ = Describe("managedJob create command", func() {
...
+			mockClient.EXPECT().CreateJob(gomock.Any(), trueClusterId, gomock.Any()).Return(fakeResp, nil).AnyTimes()
+			mockClient.EXPECT().GetRun(gomock.Any(), trueClusterId, gomock.Eq("jid")).Return(fakeJobResp, nil).AnyTimes()
+
+			sut.SetArgs([]string{"create", "SREP/something", "--cluster-id", testClusterId, "--url", "https://newbackplane.url", "--wait"})
+			err := sut.Execute()
+
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should timeout if job waiting in pending status for long time", func() {
```
Comment (bmeng) [inline cmd/ocm-backplane/managedJob/createManagedJob_test.go:201]: The title needs to be updated？

**Comment by bmeng** [comment]
Thanks! /lgtm

**Pull request merged**